homeassistant/components/rainbird/switch.py

```diff
@@ -64,6 +71,18 @@ def start_irrigation(service):
         schema=SERVICE_SCHEMA_IRRIGATION,
     )
 
+    def set_rain_delay(service):
+        duration = service.data[ATTR_DURATION]
+
+        controller.set_rain_delay(duration)
+
+    hass.services.register(
```

Comment:
This service won't work with more than one controller since we'll overwrite the service handler each time the platform is set up. To properly implement this service we need to resolve what controller to use based on some service data parameter.

It's also unclear why the switch platform is used to register the service. The switch entity is not involved in the service handler.